Sources/Client/Model/Channel/Channel.swift

```diff
@@ -196,9 +199,40 @@ public final class Channel: Codable {
         unreadMessageReadAtomic.set(messageRead)
         unreadCountAtomic.set(.noUnread)
     }
+}
+
+extension Channel: Hashable, CustomStringConvertible {
+    
+    public static func == (lhs: Channel, rhs: Channel) -> Bool {
+        lhs.cid == rhs.cid
+    }
+    
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(cid)
+    }
     
-    public func subscribe(forEvents eventTypes: Set<EventType> = Set(EventType.allCases),
+    public var description: String {
+        let opaque: UnsafeMutableRawPointer = Unmanaged.passUnretained(self).toOpaque()
+        return "Channel<\(opaque)>:\(cid):\(name ?? "<NoName>")"
+    }
+}
+
+// MARK: Subscriptions
+
+extension Channel {
+    
+    public func subscribe(forEvents eventTypes: Set<EventType> = EventType.channelCases,
                           _ callback: @escaping Client.OnEvent) -> Cancellable {
+        if eventTypes != EventType.channelCases, !eventTypes.isStrictSubset(of: EventType.channelCases) {
+            var badEvents = eventTypes
+            badEvents.subtract(EventType.channelCases)
+            
+            let message = "The events \(badEvents) are not channel events and will never get handled by your completion handler. "
+                + "Please check the documentation on event for more information."
+            
+            Client.shared.logger?.log(message, level: .error)
+        }
+        
         let subscription = Client.shared.subscribe(forEvents: eventTypes, cid: cid, callback)
```

Comment:
Doesn't make sense to still send the events here, we should send what's valid

Reply:
Fixed

Reply:
You forgot to use `validEvents` for subscription in line
```swift
let subscription = Client.shared.subscribe(forEvents: eventTypes, cid: cid, callback)
```